libbeat/cmd/instance/beat.go

```diff
@@ -307,9 +307,15 @@ func NewBeatReceiver(settings Settings, receiverConfig map[string]interface{}, u
 
 	if settings.DisableConfigResolver {
 		config.OverwriteConfigOpts(obfuscateConfigOpts())
-	} else {
+	} else if store != nil {
 		// TODO: Allow the options to be more flexible for dynamic changes
 		config.OverwriteConfigOpts(configOpts(store))
+	} else {
+		config.OverwriteConfigOpts([]ucfg.Option{
+			ucfg.PathSep("."),
+			ucfg.ResolveEnv,
+			ucfg.VarExp,
+		})
```

Comment:
I would like feedback on if we should use a conditional approach as I've done here; or if we should instead move the `configOpts` func to another file that is controlled by the `requirefips` build tag

Reply:
I think the approach you've chosen is ok (I'm a little sad to see two adjacent directories _both_ have files named `keystore_fips.go` and `keystore_nofips.go` but I don't have a better suggestion on this pass). In addition to the comment above, a suggestion I'd make is renaming `configOpts` to `configOptsWithKeystore`, just to make the specific semantic requirement more apparent at a glance.